### PR TITLE
Import & Export Users in excels

### PR DESCRIPTION
**Pull request by EvonneNgXinYi**
Construct importing and exporting users system in the excels form.

### CODE REVIEW

**Comment by EvonneNgXinYi** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -50,6 +57,18 @@ def destroy
     redirect_to users_path
   end
 
+
+  def import
+    result = User.import(params[:attachment])
+    unless result == false
```
Comment (manhnt-0960) [inline app/controllers/users_controller.rb]: why don't you use if instead of unless ?

File: app/models/user.rb [hunk 2]
```diff
@@ -86,6 +86,24 @@ def mailboxer_email(object)
     self.email
   end
 
+  def self.import file
```
Comment (manhnt-0960) [inline app/models/user.rb]: use class << sefl to define class method

Comment (EvonneNgXinYi) [reply app/models/user.rb]: Done

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (EvonneNgXinYi) [reply app/controllers/users_controller.rb]: done

**Comment by EvonneNgXinYi** [comment]
ready

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -50,6 +57,18 @@ def destroy
     redirect_to users_path
   end
 
+
+  def import
+    result = User.import(params[:attachment])
+    if result == false
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
    if result.blank?
```

File: app/models/user.rb [hunk 4]
```diff
@@ -40,6 +40,24 @@ def digest string
     def new_token
       SecureRandom.urlsafe_base64
     end
+
+    def import file 
+      spreadsheet = open_spreadsheet(file)
+      header = spreadsheet.row(1)
```
Comment (saobv-2223) [inline app/models/user.rb]: you can remove ()

File: app/models/user.rb [hunk 5]
```diff
@@ -40,6 +40,24 @@ def digest string
...
+      (2..spreadsheet.last_row).each do |i|
+        row = Hash[[header, spreadsheet.row(i)].transpose]
+        user = find_by_id(row["id"]) || new
+        user.attributes = row.to_hash.merge(password: "123456", password_confirmation: "123456")
+        user.save!
+      end
+    end
+
+    def open_spreadsheet(file)
```
Comment (saobv-2223) [inline app/models/user.rb]: you can remove (*)

File: app/views/users/index.xlsx.axlsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+wb = xlsx_package.workbook
+
+wb.styles do |s|
+  title_style = s.add_style :sz => 20, :alignment => { :horizontal=> :center }, :border => { :style => :thick, :color => "000000", :edges => [:left, :right, :top, :bottom] }
+  wrap_text = s.add_style({:alignment => {:horizontal => :center, :vertical => :center, :wrap_text => true}})
+  border_text = s.add_style(:border => { :style => :thin, :color => "000000", :edges => [:left, :right, :top, :bottom] })
```
Comment (saobv-2223) [inline app/views/users/index.xlsx.axlsx]: You should using new syntax for symbol.
For example:
`sz: 20, alignment: {horizontal: :center}`

Comment (EvonneNgXinYi) [reply app/views/users/index.xlsx.axlsx]: done

File: app/models/user.rb (hunk 5 above)
Comment (EvonneNgXinYi) [reply app/models/user.rb]: done

File: app/models/user.rb (hunk 4 above)
Comment (EvonneNgXinYi) [reply app/models/user.rb]: done

**Comment by EvonneNgXinYi** [comment]
image
image
image
Sorry. I had a problem.
When i click **Import** without choosing any file, the errors occured. 
May i know how to solve it.

**Comment by EvonneNgXinYi** [comment]
ready

**Comment by ngocvt-0484** [comment]
please debug by using byebug to know why it wrong

**Comment by ngocvt-0484** [comment]
commented

**Pull request merged**